### PR TITLE
SYNTH-4267: fail if test does not support tunnel

### PR DESCRIPTION
**Pull request by kalinichenko**
### What and why?

At the moment only Browser and HTTP tests works with the tunnel. Those that have no support, wrongly logged as ones running through the tunnel and might result in a false positive.

### How?

If specified the test which does not support to run with the tunnel, fail the job.

### CODE REVIEW

File: src/commands/synthetics/utils.ts [hunk 1]
```diff
@@ -514,13 +515,19 @@ export const getTestAndOverrideConfig = async (
...
 
-    return {overriddenConfig, test}
+    return {overriddenConfig}
+  }
+  reporter.testWait(test)
+
+  if (isTunnelEnabled && test.type !== 'browser' && test.subtype !== 'http') {
+    throw new CriticalError(
+      'TUNNEL_NOT_SUPPORTED',
+      `${test.subtype || test.type} tests do not support to run with the tunnel (public_id: ${test.public_id})`
```
Comment (ruizb) [inline src/commands/synthetics/utils.ts:526]:
```suggestion
      `The tunnel is only supported with HTTP API tests and Browser tests (public ID: ${test.public_id}, type: ${
        test.type
      }${test.subtype ? `, sub-type: ${test.subtype}` : ''}).`
```

File: src/commands/synthetics/__tests__/utils.test.ts [hunk 2]
```diff
@@ -397,20 +397,54 @@ describe('utils', () => {
...
 
-    test('Forbidden error when getting a test', async () => {
       const triggerConfig = {suite: 'Suite 1', config: {}, id: '123-456-789'}
 
       await expect(() =>
         utils.getTestAndOverrideConfig(api, triggerConfig, mockReporter, getSummary())
       ).rejects.toThrow('Failed to get test: could not query https://app.datadoghq.com/example\nForbidden\n')
     })
+
+    test('Passes when a tunnel is enabled for HTTP test', async () => {
```
Comment (ruizb) [inline src/commands/synthetics/__tests__/utils.test.ts:413]:
```suggestion
    test('Passes when the tunnel is enabled for HTTP test', async () => {
```

File: src/commands/synthetics/__tests__/utils.test.ts [hunk 3]
```diff
@@ -397,20 +397,54 @@ describe('utils', () => {
...
+        return {data: {subtype: 'http', public_id: '123-456-789'}}
+      }) as any)
+
+      const triggerConfig = {suite: 'Suite 1', config: {}, id: '123-456-789'}
+      expect(await utils.getTestAndOverrideConfig(api, triggerConfig, mockReporter, getSummary(), true)).toEqual(
+        expect.objectContaining({test: expect.objectContaining({public_id: '123-456-789', subtype: 'http'})})
+      )
+    })
+
+    test('Passes when a tunnel is enabled for Browser test', async () => {
```
Comment (ruizb) [inline src/commands/synthetics/__tests__/utils.test.ts:425]:
```suggestion
    test('Passes when the tunnel is enabled for Browser test', async () => {
```

File: src/commands/synthetics/__tests__/utils.test.ts [hunk 4]
```diff
@@ -397,20 +397,54 @@ describe('utils', () => {
...
+        return {data: {type: 'browser', public_id: '123-456-789'}}
+      }) as any)
+
+      const triggerConfig = {suite: 'Suite 1', config: {}, id: '123-456-789'}
+      expect(await utils.getTestAndOverrideConfig(api, triggerConfig, mockReporter, getSummary(), true)).toEqual(
+        expect.objectContaining({test: expect.objectContaining({public_id: '123-456-789', type: 'browser'})})
+      )
+    })
+
+    test('Fails when getting a test not supporting a tunnel', async () => {
```
Comment (ruizb) [inline src/commands/synthetics/__tests__/utils.test.ts:437]:
```suggestion
    test('Fails when the tunnel is enabled for an unsupported test type', async () => {
```

**Review by ruizb — APPROVED** [review]
Just some nits, LGTM otherwise 🐰

**Comment by blhblah1** [comment]
I've been running multi-step tests with v2.2.0 and they have been working correctly. Please bring back this functionality!

**Pull request merged**